core/commands/ping.go

```diff
@@ -0,0 +1,148 @@
+package commands
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"time"
+
+	cmds "github.com/jbenet/go-ipfs/commands"
+	peer "github.com/jbenet/go-ipfs/p2p/peer"
+	u "github.com/jbenet/go-ipfs/util"
+
+	context "github.com/jbenet/go-ipfs/Godeps/_workspace/src/code.google.com/p/go.net/context"
+)
+
+const kPingTimeout = 10 * time.Second
+
+type PingResult struct {
+	Success bool
+	Time    time.Duration
+	Text    string
+}
+
+var PingCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "send echo request packets to IPFS hosts",
+		Synopsis: `
+ipfs ping <peer.ID> - Send pings to a peer using the routing system to discover its address
```

Comment:
Synopsis doesn't need the `ipfs ping <peer.ID>` part. try it out with `ipfs ping -h`, which will show the synopsys.

```Go
		Synopsis: `
Send pings to a peer. If not provided, the routing system to discover the peer's address.
`
```

yeah that o/ implies we should allow:

```
ipfs ping /ip4/1.2.3.4/tcp/1245/Qm.....
```